src/Saga/Saga.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.Extensions.Logging;
+
+namespace TemporalioSamples.Saga;
+
+public class Saga
```

Comment:
I think this may be adding more features than is needed. I think this should follow the TypeScript sample, see https://github.com/temporalio/samples-typescript/tree/main/saga. Basically just put a `List<Func<Task>> compensations = new()` at the top of the function needing compensations (i.e. the workflow function), then `foreach (var comp in compensations.Reverse() {` in the `catch`. I don't think on-compensation-complete or on-compensation-error callbacks are needed in the sample either, users can implement those as they need in their code. Users may have different things they need to do (or not do) in each situation.

I think there's value in showing that "saga" is not a special thing in Temporal, anyone can run any code in catch. There's no need to break this out.

Reply:
@cretz fixed